### ISSUE TITLE
Missing multicall address for chain id 31337 

### ISSUE THREAD

**Issue by ironsoul0** [EthWorks/useDApp#267]
Hi team!

Thank you for your amazing work. I just faced the issue while trying to develop on a local chain using hardhat.

I keep getting this error on a console every time I try to send some transaction or connect my wallet.

![image](https://user-images.githubusercontent.com/31582693/122636629-d4897600-d11c-11eb-99f2-05ecadc619c2.png)

My Metamask points to local hardhat network `localhost:8545` and I configured `chainId` (in Metamask) for this network to be `31337`. So it now matches with `useDApp` hardhat's `chainId`. However, this did not help.

May I ask for any help if possible? Thanks!

**Comment by tankvt** [comment]
Hi @ironsoul0,

I updated my Config as below and it worked for me

```
import {
    ChainId,
    Config,
    DAppProvider,
    MULTICALL_ADDRESSES,
} from "@usedapp/core";

const config: Config = {
    readOnlyUrls: {
        31337: "http://localhost:8545",
    },
    multicallAddresses: {
        31337: "YOUR_LOCAL_ADDRESS_HERE", 
        ...MULTICALL_ADDRESSES,
    },
    supportedChains: [
        ChainId.Mainnet,
        ChainId.Goerli,
        ChainId.Kovan,
        ChainId.Rinkeby,
        ChainId.Ropsten,
        ChainId.xDai,
        ChainId.BSC,
        ChainId.Localhost,
        ChainId.Hardhat,
    ],
};
```

Hope that help you.

**Comment by ChangoMan** [comment]
> Hi @ironsoul0,
> 
> I updated my Config as below and it worked for me
> 
> ```
> import {
>     ChainId,
>     Config,
>     DAppProvider,
>     MULTICALL_ADDRESSES,
> } from "@usedapp/core";
> 
> const config: Config = {
>     readOnlyUrls: {
>         31337: "http://localhost:8545",
>     },
>     multicallAddresses: {
>         31337: "YOUR_LOCAL_ADDRESS_HERE", 
>         ...MULTICALL_ADDRESSES,
>     },
>     supportedChains: [
>         ChainId.Mainnet,
>         ChainId.Goerli,
>         ChainId.Kovan,
>         ChainId.Rinkeby,
>         ChainId.Ropsten,
>         ChainId.xDai,
>         ChainId.BSC,
>         ChainId.Localhost,
>         ChainId.Hardhat,
>     ],
> };
> ```
> 
> Hope that help you.

@tankvt Thanks for sharing your config! May I ask how you deployed a multicall contract on your local chain? Thanks!

**Comment by tankvt** [comment]
Hi @ChangoMan,

I saw the PR https://github.com/EthWorks/useDApp/pull/245 merged to the latest version. 
You can get the latest version and check it again.

Regards,

**Comment by ChangoMan** [comment]
@tankvt Thanks for the reply! I updated to the newest version but ran into a bug when using it with Next.js. I've opened a PR (https://github.com/EthWorks/useDApp/pull/277) to fix this.

I've fixed it locally so I'll try to get the multicall working now.

**Comment by tt-marek** [comment]
Merged in #245

**Comment by okorieebube** [comment]
Guys, i believe you built a great tool, but i must say the documentation isn't helpful atall. 
I keep running into issues with this multicall feature, and i cant find a way to fix it in the documentation.
The docs doesnt have great step by step directives or even other online resources explaining how to setup even a basic project connected to your local blockchain
Its frustrating.
